src/sunmatrix/sparse/sunmatrix_sparse.c

```diff
@@ -225,51 +215,52 @@ SUNMatrix SUNSparseFromDenseMatrix(SUNMatrix Ad, sunrealtype droptol,
  * if the request for matrix storage cannot be satisfied.
  */
 
-SUNMatrix SUNSparseFromBandMatrix(SUNMatrix Ad, sunrealtype droptol,
+SUNMatrix SUNSparseFromBandMatrix(SUNMatrix Ab, sunrealtype droptol,
                                   int sparsetype)
 {
+  SUNFunctionBegin(Ab->sunctx);
   sunindextype i, j, nnz;
   sunindextype M, N;
   SUNMatrix As;
 
-  /* check for legal sparsetype, droptol and input matrix type */
-  if ((sparsetype != CSR_MAT) && (sparsetype != CSC_MAT)) { return NULL; }
-  if (droptol < ZERO) { return NULL; }
-  if (SUNMatGetID(Ad) != SUNMATRIX_BAND) { return NULL; }
+  SUNAssertNull(SUNMatGetID(Ab) == SUNMATRIX_BAND, SUN_ERR_ARG_OUTOFRANGE);
+  SUNAssertNull(sparsetype == CSC_MAT || sparsetype == CSR_MAT,
+                SUN_ERR_ARG_OUTOFRANGE);
```

Comment:
Like above, should be `SUN_ERR_ARG_INCOMPATIBLE` or `SUN_ERR_ARG_WRONGTYPE` instead?